### DISCUSSION TITLE
Pin GitHub action digests, but only for released versions.

### DISCUSSION THREAD

**Discussion by datosh** [renovatebot/renovate#18282] [Request Help] [answered]
I would like to pin my GitHub action dependencies, but only receive released versions. Is this possible with the current renovate configuration options or should I create an issue (feature request) instead?

## Here is a short example

I rely on `actions/checkout` and use version v3.0.2 pinned via digest:

```yaml
    - name: Check out repository
      uses: actions/checkout@2541b1294d2704b0964813337f33b291d3f8596b # v3.0.2
```

I would expect renovate to upgrade my dependency to the next released version.
(best case it also updates the comment in one go)

```yaml
    - name: Check out repository
      uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # v3.1.0
```

Instead I am updated to the [latest commit on the default (main) branch](https://github.com/actions/checkout/commit/8230315d06ad95c617244d2f265d237a1682d445):

```yaml
    - name: Check out repository
      uses: actions/checkout@8230315d06ad95c617244d2f265d237a1682d445
```

I don't think that RegEx would help me here, since there is no standard naming schema to release GitHub actions, as you would find in other ecosystems/package-managers, e.g., pip.

**Comment by viceice** [comment]
not sure what you are doing, but renovate has a buildin pin GitHub actions function.

the comment needs to be a little different. checkout this repo for sample.

**Reply by datosh** [reply, marked as answer]
Thanks got it working! Being a new renovate user, it wasn't immediately clear to me what the correct configuration was. Looking at other repository configs helped.
